install/linux.md

```diff
@@ -81,9 +81,7 @@ Hier müssen im Terminal die folgenden Zeilen eingegeben werden:
 ```bash
 curl -L -O "https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-$(uname)-$(uname -m).sh"
 bash Mambaforge-$(uname)-$(uname -m).sh -p ~/.local/mambaforge -b
-source "${HOME}/.local/mambaforge/etc/profile.d/conda.sh"
 source "${HOME}/.local/mambaforge/etc/profile.d/mamba.sh"
-mamba activate
 mamba init ${0//-}
```

Comment:
Any reason to use this magic instead of `$SHELL`?

Reply:
Interesting.
For me `${0//-}` actually just returns `zsh` and `$(basename ${0//-})` returns nothing. So that wouldn't work in my case.
I guess the idea of using $0 instead of $SHELL is installing to the current shell instead of the underlying base shell.
For some cases these can differ and that could cause installation issues

Reply:
Indeed, `$SHELL` is wrong when e.g. starting bash in zsh, still, the basename is needed for me:

```
❯ echo $SHELL
/bin/zsh
❯ echo ${0//-}
/bin/zsh
❯ echo $(basename ${0//-})
zsh
❯ /bin/bash
(base) mnoethe@e5b-dell-12:~$ echo $SHELL
/bin/zsh
(base) mnoethe@e5b-dell-12:~$ echo ${0//-}
/bin/bash
(base) mnoethe@e5b-dell-12:~$ echo $(basename ${0//-})
bash
```

Reply:
> `${ps -p "$$" -o command=""}`

That fails for me with both zsh and bash with a "bad substitution" error

Reply:
```
❯ ps -p "$$" -o command=""
/bin/zsh
```

so also here, the basename would be needed

Reply:
I guess it depends whether your shell is an absolute path or just the program:

```
❯ getent passwd mnoethe
mnoethe:*:5074:5002:Maximilian Linhoff:/home/mnoethe:/bin/zsh
```

Reply:
That might be it.
```
➜  ~ getent passwd luetge 
luetge:*:1142:1005:Sebastian Luetge:/home/luetge:/bin/bash
```
I start an active zsh session in a bash shell, which returns the program but in the default shell the return could be /bin/bash

The shell in /etc/passwd is also absolute but again bash instead of zsh

How can we sort that for "easy to execute" instructions without knowledge on the active shell?

Reply:
Or we say mamba init --all

Reply:
At least for all supported shells it should be fine and is less magic

Reply:
> You should fix your system.... the proper way to use zsh as your shell is to set it as your login shell `chsh -s /bin/zsh luetge`

In theory - yes. That is not the only wonky part with my setup on this machine. But that's a different story altogether ;)

Reply:
We shouldn't make the installation instructions here bullet proof for every edge case, it overcomplicates it for the normal users.

Reply:
The most important ones are Ubuntu, and WSL on 10 and 11, I guess.

Reply:
On Win10 and Win11 I dont use the flags for the install script.
The installer asks you to run `conda init` in the last step. Seems to work fine.